src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,79 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY_OPERATION_TYPE = "supply";
+    private static final String BUY_OPERATION_TYPE = "buy";
+    private static final String RESULT_OPERATION_TYPE = "result";
+    private static final String[] OPERATION_TYPE_FIELDS =
+            {SUPPLY_OPERATION_TYPE, BUY_OPERATION_TYPE, RESULT_OPERATION_TYPE};
+    private static final int SUPPLY_AMOUNT_TYPE_INDEX = 0;
+    private static final int BUY_AMOUNT_TYPE_INDEX = 1;
+    private static final int RESULT_AMOUNT_TYPE_INDEX = 2;
+    private static final int FIELD_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFile(fromFileName);
+        int[] amountTypeFields = new int[OPERATION_TYPE_FIELDS.length];
+        calculateValuesForAmountTypeFields(amountTypeFields, dataFromFile);
+        String report = createReport(amountTypeFields);
+        writeFile(toFileName, report);
```

Comment:
```suggestion
        String dataFromFile = readFile(fromFileName);
        String report = createReport(dataFromFile);
        writeFile(toFileName, report);
```

Reply:
In common mistakes it is written that methods should be short, between 5-15 lines. Now my method createReport() will be very big. In addition if there is no array, I have to fill stringbuilder manually. I thought it is better with for() cycle.